example_generate_hosts.py

```diff
@@ -0,0 +1,38 @@
+"""
+Hosts Generation
+
+Input: my[001..900].domain.com
+
+Output: my001.domain.com, my002.domain.com, ... my900.domain.com
+"""
+
+def generate_hosts(inventory):
+    """
+    Function that will generates a list of hosts based
+    in the specific string 'my[001..900].domain.com'
+    
+    :param inventory: specific string
+
+    returns a list with hosts names.  
+    """
+    prefix = inventory.split('[')[0]
+    first = inventory.replace('..', '[', 1).split('[')[1]
+    last = inventory.replace('..', ']', 1).split(']')[1]
+    domain = inventory.replace('.', ' ', 3).split(' ')[-1]
+
+    hosts = []
+
+    for x in range(int(first), int(last) + 1):
+        
+        if len(str(x)) == 1:
```

Comment:
The string size should be given by the length of `first`. Example: `my[001..10].domain.com` should generate from `my001.domain.com` to `my010.domain.com`. Then you can format the string with `{x:0{size}d}`.